robusta_krr/core/integrations/kubernetes.py

```diff
@@ -40,30 +41,35 @@ async def list_scannable_objects(self) -> list[K8sObjectData]:
 
         self.info(f"Listing scannable objects in {self.cluster}")
         self.debug(f"Namespaces: {self.config.namespaces}")
+        self.debug(f"Resources: {self.config.resources}")
 
         try:
-            objects_tuple = await asyncio.gather(
-                self._list_deployments(),
-                self._list_all_statefulsets(),
-                self._list_all_daemon_set(),
-                self._list_all_jobs(),
-            )
+            list_fns = []
+            if self._should_list_resource("deployments"):
+                list_fns.append(self._list_deployments)
+            if self._should_list_resource("statefulsets"):
+                list_fns.append(self._list_all_statefulsets)
+            if self._should_list_resource("daemon_set"):
+                list_fns.append(self._list_all_daemon_set)
+            if self._should_list_resource("jobs"):
+                list_fns.append(self._list_all_jobs)
+
```

Comment:
We should add a warning in case filtering by resources ends up in filtering everything, as it might be confusing to user.